### PR TITLE
feat: incorporate manager entity in vault

### LINKED ISSUE #133: Incorporate manager entity in vault
We want to have more granularity on gov only methods.
The idea is to keep the following methods gov only

```
setName
setSymbol
setGovernance
acceptGovernance
setGuestList
setRewards
sweep
setPerformanceFees
setManagementFees
addStrategy
updateStrategyPerformanceFee
migrateStrategy
setDepositLimit
```

and make the following manager +gov available:

```
setWithdrawalQueue
updateStrategyDebtLimit
updateStrategyRateLimit
addStrategyToQueue
removeStrategyFromQueue
```

### PR DESCRIPTION
**Pull request by Macarse**
Fixes #133

### CODE REVIEW

File: tests/functional/vault/test_management.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import pytest
```
Comment (fubuloubu) [inline tests/functional/vault/test_management.py:1]: Can you add these test additional checks to the other cases in the test suite instead?

**Review by fubuloubu — APPROVED** [review]
A few suggestions to change, otherwise looks good

File: contracts/Vault.vy [hunk 2]
```diff
@@ -507,7 +529,7 @@ def setWithdrawalQueue(queue: address[MAXIMUM_STRATEGIES]):
         The array of addresses to use as the new withdrawal queue. This is
         order sensitive.
     """
-    assert msg.sender == self.governance
+    assert msg.sender in [self.governance, self.management]
```
[the remark this answers is not in the document]
Comment (Macarse) [reply contracts/Vault.vy:532]: great tip! fixed!

File: tests/functional/vault/test_management.py (hunk 1 above)
Comment (Macarse) [reply tests/functional/vault/test_management.py]: fixed!

File: tests/functional/vault/test_config.py [hunk 3]
```diff
@@ -43,32 +45,174 @@ def test_vault_name_symbol_override(guardian, gov, rewards, token, Vault):
...
-        ("depositLimit", "setDepositLimit", 1000),
+        ("name", "setName", "NewName yVault", True, False, False),
+        ("symbol", "setSymbol", "yvNEW", True, False, False),
+        ("emergencyShutdown", "setEmergencyShutdown", True, True, False, True),
+        ("guardian", "setGuardian", None, True, False, True),
+        ("rewards", "setRewards", None, True, False, False),
+        ("management", "setManagement", None, True, False, False),
+        ("performanceFee", "setPerformanceFee", 1000, True, False, False),
+        ("managementFee", "setManagementFee", 1000, True, False, False),
+        ("depositLimit", "setDepositLimit", 1000, True, False, False),
```
Comment (fubuloubu) [inline tests/functional/vault/test_config.py]: Doesn't seem like management is used at all in these methods. Perhaps delete it?

File: tests/functional/vault/test_config.py [hunk 4]
```diff
@@ -43,32 +45,174 @@ def test_vault_name_symbol_override(guardian, gov, rewards, token, Vault):
...
 
 
+@pytest.mark.parametrize(
+    "key,setter,val,gov_allowed,mgmt_allowed,guard_allowed",
+    [
+        ("debtLimit", "updateStrategyDebtLimit", 500, True, True, False),
+        ("rateLimit", "updateStrategyRateLimit", 10, True, True, False),
+    ],
+)
+def test_vault_updateStrategy(
```
Comment (fubuloubu) [inline tests/functional/vault/test_config.py:110]: Guardian is never allowed on these functions, and gov/mgmt can always both call them. Perhaps not necessary here.

File: tests/functional/vault/test_config.py [hunk 5]
```diff
@@ -43,32 +45,174 @@ def test_vault_name_symbol_override(guardian, gov, rewards, token, Vault):
...
+
+@pytest.mark.parametrize(
+    "count,setter,gov_allowed,mgmt_allowed,guard_allowed",
+    [
+        (20, "setWithdrawalQueue", True, True, False),
+        (0, "removeStrategyFromQueue", True, True, False),
+        (1, "addStrategyToQueue", True, True, False),
+    ],
+)
+def test_vault_withdrawalQueue(
```
Comment (fubuloubu) [inline tests/functional/vault/test_config.py:158]: Same as above

File: tests/functional/vault/test_config.py (hunk 3 above)
Comment (fubuloubu) [reply tests/functional/vault/test_config.py]: Also note that governance is always allowed, so that flag doesn't do anything

Comment (Macarse) [reply tests/functional/vault/test_config.py]: fixed!

**Review by fubuloubu — APPROVED** [review]

File: contracts/Vault.vy [hunk 6]
```diff
@@ -120,12 +121,16 @@ event UpdateGovernance:
     governance: address # New active governance
 
 
+event UpdateManagement:
+    management: address # New active governance
```
Comment (fubuloubu) [inline contracts/Vault.vy:125]:
```suggestion
    management: address # New active manager
```

**Pull request merged**